src/services/validatorSchema.ts

```diff
@@ -0,0 +1,160 @@
+import * as yup from "yup";
+
+export const UserSchema = yup.object({
+    user_id: yup
+        .string()
+        .strict()
+        .typeError("user_id must be a string.")
+        .defined("The user_id (string) must be provided.")
+});
+
+export const ItemSchema = yup.object({
+    item_id: yup
+        .number()
+        .typeError("item_id must be an integer.")
+        .positive("item_id must be a positive integer.")
+        .defined("The item_id (positive integer) must be provided.")
+});
+
+export const InventorySchema = yup.object({
+    inventory_item_id: yup
+        .string()
+        .strict()
+        .typeError("inventory_item_id must be a string.")
+        .defined("The inventory_item_id (string) must be provided.")
+});
+
+export const DataSchema = yup.object({
+    data: yup.lazy((value) => {
+        if (value === undefined || value === null)
+            return yup
+                .object()
+                .required("The data (object of string(s)) must be provided.");
+        else {
+            const schema = Object.keys(value).reduce(
+                (acc: any, curr: string) => {
+                    acc[curr] = yup
+                        .string()
+                        .strict()
+                        .typeError("data's field must be a string.")
+                        .required(
+                            "The data's field (string) must be provided."
+                        );
+                    return acc;
+                },
+                {}
+            );
+            return yup
+                .object()
+                .shape(schema)
+                .required("The data (object of string(s)) must be provided.");
+        }
+    })
+});
+
+export const JsonUserSchema = yup.object({
+    user_id: yup
+        .number()
+        .typeError("user_id must be an integer.")
+        .positive("user_id must be a positive integer.")
+        .defined("The user_id (positive integer) must be provided."),
+    inventory_address: yup
+        .string()
+        .strict()
+        .typeError("inventory_address must be a string.")
+        .defined("The inventory_address (string) must be provided.")
+});
+
+export const ItemSpecsSchema = yup.object({
+    start: yup
+        .number()
+        .typeError("start must be an integer.")
+        .positive("start must be a positive integer.")
```

Comment:
start can definitely be `0` but yup doesn't consider `0` to be "positive". This requesT:

```
curl --location --request GET 'http://localhost:8000/api/item-store/item?start=0&limit=10'
```

returns this error:

```
{
    "message": "start must be a positive integer."
}
```

can you check?